src/common/config.cc

```diff
@@ -689,7 +689,7 @@ int md_config_t::injectargs(const std::string& s, std::ostream *oss)
   char *p = b;
   while (*p) {
     nargs.push_back(p);
-    while (*p && *p != ' ') p++;
+    while (*p != ' ') p++;
```

Comment:
yeah, it's not equivalent, but it seems to have nothing to do with that.
As (*p != '\0' && *p != ' ') is always running after while(*p) , so " *p != '\0' " is always true. there is no necessary to judge whether *p != '\0' is true again.